internal/asm/...: add documentation

kortschak opened the pull request:
Please take a look.

This may seem a little odd, but it arose out of a discussion I was having with someone who wanted to contribute to the complex blas implementation. These comments will help people working on the blas and floats packages

Fixes #47.

kortschak committed:
internal/asm/...: add documentation

This documentation is intended to allow gonum developers to see exactly
what an asm function is intended to do. They are not for external code
consumers.

vladimir-ch commented on asm/c128/doc.go:
```diff
@@ -0,0 +1,6 @@
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package c128 provides complex128 vector primatives.
```
primitives here and in the other packages

kortschak replied:
Oops. Fixed.

vladimir-ch commented on asm/f64/stubs_amd64.go:
```diff
@@ -6,42 +6,160 @@
+//  	norm += math.Abs(t[i] - v)
+//  }
+//  return norm
 func L1Norm(s, t []float64) float64
```
These two should be called L1Dist (or L1Distance) and LinfDist.

kortschak replied:
Done.

vladimir-ch commented on asm/f64/stubs_noasm.go:
```diff
@@ -8,32 +8,59 @@ package f64
+//  	sum += math.Abs(v)
+//  }
+//  return sum
 func AbsSum(x []float64) (sum float64) {
```
This one could be called L1Norm.

kortschak replied:
Done.

kortschak committed:
internal/asm/...: rename functions

These are unused outside the asm packages and so are safe to rename now.

L1Norm => L1Dist
LinfNorm => LinfDist
AbsSum => L1Norm

kortschak commented:
PTAL

Kunde21 reviewed (approved):
LGTM
